pkg/template/nstemplatetier_test.go

```diff
@@ -0,0 +1,95 @@
+package template_test
+
+import (
+	"testing"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/member-operator/pkg/apis"
+	"github.com/codeready-toolchain/member-operator/pkg/template"
+	"github.com/codeready-toolchain/toolchain-common/pkg/cluster"
+
+	"github.com/operator-framework/operator-sdk/pkg/log/zap"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+func TestGetNSTemplateTier(t *testing.T) {
+
+	// given
+	// Setup Scheme for all resources (required before adding objects in the fake client)
+	err := apis.AddToScheme(scheme.Scheme)
+	require.NoError(t, err)
+	logf.SetLogger(zap.Logger())
+	basicTier := &toolchainv1alpha1.NSTemplateTier{
```

Comment:
Let's have multiple tiers in the client available. Not the only one. And the NS types should use the same names in different tiers.
Like `basicTier` with `stage`, `cicd`, `ide`. And `advanced` with its own `stage`, `cicd`, `ide`. So we can make sure that the templates loaded from the correct tier.

Reply:
ok, let me do that

Reply:
done in 66d2d6d